packages/extension/src/shared/discovery.ts

```diff
@@ -0,0 +1,66 @@
+import { ExplorerProvider, TOTAL_NUMBER_OF_GROUPS } from '@alephium/web3'
+import { WalletAccount } from './wallet.model'
+
+const minGap = 5
+const derivationBatchSize = 10
+
+export abstract class AccountDiscovery {
```

Comment:
I am a bit lost in the discovery mechanism, could you describe the approach in simple sentences?

Reply:
Here's how it works:

1. For each group, start deriving accounts from hd index 0, deriving 10 accounts at a time
2. Then check if these addresses are active. Stop deriving if the last five accounts are inactive; otherwise, continue deriving accounts from the latest hd index
3. The wallet will only store active accounts.

Correct me if I'm wrong @h0ngcha0

Reply:
The current discovery mechanism starts deriving from 0 for each group, which can lead to duplicate calculations. I have modified the discovery mechanism to derive accounts from index 0 and stop deriving when each group has 5 consecutive inactive addresses.